### PR TITLE
16x tests is producing wrong results nan's appear after the all_reduce call

### PR DESCRIPTION
**Pull request by kalyanjk**
16x tests is producing wrong results nan's appear after the all_reduce call

### CODE REVIEW

File: optimum/habana/transformers/models/llama/modeling_llama.py [hunk 1]
```diff
@@ -407,7 +407,7 @@ def pre_attn_forward(
             # to avoid making past key values as persistent output tensors of HPU graphs.
             past_key_value = (past_key_value[0].shape, past_key_value[1].shape)
 
-        return attn_output, attn_weights, past_key_value
+        return attn_output + 0, attn_weights, past_key_value
```
Comment (libinta) [inline optimum/habana/transformers/models/llama/modeling_llama.py:410]: can you explain why we need to change model code to fix the allreduce nan issue?

Comment (mandy-li) [reply optimum/habana/transformers/models/llama/modeling_llama.py:410]: Same question. why this fix allreduce nan issue? Did you test it on 16x?

Comment (kalyanjk) [reply optimum/habana/transformers/models/llama/modeling_llama.py:410]: There are back to back all_reduce calls 
one in attn_output = self.o_proj(attn_output)
and followed by line 533 self.self_attn.attention_all_reduce(hidden_states)
if attn_output is not used between these two all_reduce calls its becoming nan
This is a temporary fix.

Comment (dvarshney-habana) [reply optimum/habana/transformers/models/llama/modeling_llama.py:410]: @mandy-li  @libinta  : To be clear  - its a workaround which we believe is avoiding a sync issue in our stack. We are not planning to merge it at the moment. 
@kalyanjk : do you know if there  is any perf impact on already working configurations ?

**Comment by kalyanjk** [comment]
This patch was experimental. will not be merged

**Pull request closed without merging**